File: teahouse/teahouse/class/tool/searchButton.m
//
//  searchButton.m
//  teahouse
//
//  Created by yuxin on 2016/12/29.
//  Copyright © 2016年 yuxin. All rights reserved.
//

#import "searchButton.h"

@implementation searchButton
- (instancetype)init {
    self = [super init];
    if (self) {
        [self.titleLabel setFont:[UIFont systemFontOfSize:14]];
    }
    return self;
}

- (CGRect)imageRectForContentRect:(CGRect)contentRect {
    return CGRectMake(5, 5, 20, 20);
}

- (CGRect)titleRectForContentRect:(CGRect)contentRect {
    CGFloat titleX = self.imageView.frame.size.width + 10;
    CGFloat titleW = contentRect.size.width - self.imageView.frame.size.width - 10;
    return CGRectMake(titleX, 5, titleW, 20);
}

@end

File: teaHouse_swift/teaHouse_swift/class/tool/searchButton.swift
//
//  searchButton.swift
//  teaHouse_swift
//
//  Created by yuxin on 2017/3/1.
//  Copyright © 2017年 yuxin. All rights reserved.
//

import UIKit

class searchButton: UIButton {

    override func imageRect(forContentRect contentRect: CGRect) -> CGRect {
        return CGRect(x: 5, y: 5, width: 20, height: 20)
    }
    
    override func titleRect(forContentRect contentRect: CGRect) -> CGRect {
        let titleX = (self.imageView?.frame.size.width)! + 10
        let titleW = contentRect.size.width - (self.imageView?.frame.width)! - 10
        return CGRect(x: titleX, y: 5, width: titleW, height: 20)
    }
}


    
